src/Storages/ObjectStorageQueue/ObjectStorageQueueOrderedFileMetadata.cpp

```diff
@@ -288,79 +411,56 @@ std::pair<bool, ObjectStorageQueueIFileMetadata::FileStatus::State> ObjectStorag
 
     processor_info = getProcessorInfo(generateProcessingID());
 
+    std::string processed_node_hive_path;
+    if (is_path_with_hive_partitioning)
+        processed_node_hive_path = std::filesystem::path(processed_node_path) / getHivePart(path);
+
     const size_t max_num_tries = 100;
     Coordination::Error code;
     for (size_t i = 0; i < max_num_tries; ++i)
     {
         std::optional<NodeMetadata> processed_node;
-        Coordination::Stat processed_node_stat;
-        std::optional<std::pair<bool, ObjectStorageQueueIFileMetadata::FileStatus::State>> result;
-        zk_retry.retryLoop([&]
-        {
-            bool is_multi_read_enabled = zk_client->isFeatureEnabled(DB::KeeperFeatureFlag::MULTI_READ);
-            if (is_multi_read_enabled)
-            {
-                Coordination::Requests requests;
-                std::vector<std::string> paths{processed_node_path, failed_node_path};
-                zkutil::ZooKeeper::MultiTryGetResponse responses = ObjectStorageQueueMetadata::getZooKeeper(log)->tryGet(paths);
+        std::optional<Coordination::Stat> processed_node_stat;
 
-                auto check_code = [this](auto code_)
-                {
-                    if (!(code_ == Coordination::Error::ZOK || code_ == Coordination::Error::ZNONODE))
-                        throw zkutil::KeeperException::fromPath(code_, path);
-                };
-                check_code(responses[0].error);
-                check_code(responses[1].error);
+        Coordination::Stat processed_node_stat_;
+        auto processed_file_info = getLastProcessedFile(&processed_node_stat_, true, log);
```

Comment:
Why do we have `check_failed = true` here if `is_failed` is unused? Also `failed` state is strange in context of "last processed file" which implied that the file is processed, not failed.